src/ray/common/id.h

```diff
@@ -74,7 +74,9 @@ class BaseID {
 
  protected:
   BaseID(const std::string &binary) {
-    std::memcpy(const_cast<uint8_t *>(this->Data()), binary.data(), T::Size());
+    RAY_CHECK(binary.size() == Size() || binary.size() == 0)
```

Comment:
why do we specifically allow empty binary string here?

Reply:
Grpc strings default to "", so calling FromBinary on that would fail the check.